packages/core/src/common/utils.ts

```diff
@@ -86,21 +86,81 @@ export function countDecimalPlaces(num: number) {
 }
 
 /**
- * Throttle an event on an EventTarget by wrapping it in `requestAnimationFrame` call.
+ * Throttle an event on an EventTarget by wrapping it in a `requestAnimationFrame` call.
  * Returns the event handler that was bound to given eventName so you can clean up after yourself.
  * @see https://developer.mozilla.org/en-US/docs/Web/Events/scroll
  */
 export function throttleEvent(target: EventTarget, eventName: string, newEventName: string) {
-    let running = false;
-    /* istanbul ignore next: borrowed directly from MDN */
-    const func = (event: Event) => {
-        if (running) { return; }
-        running = true;
-        requestAnimationFrame(() => {
+    const throttledFunc = throttleHelper(
+        undefined,
+        undefined,
+        (event: Event) => {
             target.dispatchEvent(new CustomEvent(newEventName, event));
-            running = false;
+        },
+    );
+    target.addEventListener(eventName, throttledFunc);
+    return throttledFunc;
+};
+
+export interface IThrottledReactEventOptions {
+    preventDefault?: boolean;
+};
+
+/**
+ * Throttle a callback by wrapping it in a `requestAnimationFrame` call. Returns the throttled
+ * function.
+ * @see https://www.html5rocks.com/en/tutorials/speed/animations/
+ */
+export function throttleReactEventCallback(
+    callback: (event: React.SyntheticEvent<any>, ...otherArgs: any[]) => any,
+    options: IThrottledReactEventOptions = {},
+) {
+    const throttledFunc = throttleHelper(
+        (event2: React.SyntheticEvent<any>) => {
```

Comment:
There's already an `(event: React.SyntheticEvent<any>, ...otherArgs: any[]) => any` specified in the callback type. Felt this would be clearer.